packages/devtools_shared/lib/src/extensions/extension_model.dart

```diff
@@ -25,31 +25,27 @@ class DevToolsExtensionConfig {
       codePoint = codePointFromJson as int? ?? defaultCodePoint;
     }
 
-    final name = json[nameKey] as String?;
-    final path = json[pathKey] as String?;
-    final issueTrackerLink = json[issueTrackerKey] as String?;
-    final version = json[versionKey] as String?;
-
-    final nullFields = [
-      if (name == null) nameKey,
-      if (path == null) pathKey,
-      if (issueTrackerLink == null) issueTrackerKey,
-      if (version == null) versionKey,
-    ];
-    if (nullFields.isNotEmpty) {
+    if (json
+        case {
+          nameKey: final String name,
+          pathKey: final String path,
+          issueTrackerKey: final String issueTracker,
+          versionKey: final String version,
+        }) {
+      return DevToolsExtensionConfig._(
+        name: name,
+        path: path,
+        issueTrackerLink: issueTracker,
+        version: version,
+        materialIconCodePoint: codePoint,
+      );
+    } else {
+      final requiredKeys = {nameKey, pathKey, issueTrackerKey, versionKey};
       throw StateError(
-        'missing required fields ${nullFields.toString()} in the extension '
-        'config.json',
+        'Missing required fields ${requiredKeys.difference(json.keys.toSet())} '
```

Comment:
More of an FYI than anything, this won't catch the case where a key is expected but of the wrong type. Maybe we should throw a different exception if `requiredKeys.differences(json.keys.toSet()) == const {}`?

Reply:
addressed.